src/prefect/utilities/urls.py

```diff
@@ -152,7 +152,7 @@ def url_for(
     base_url = base_url or default_base_url
 
     if not base_url:
-        logger.warning(
+        logger.info(
```

Comment:
I'd vote for this to be a debug-level log. It will likely show up on a user's first run and contains no info on how severe the issue is or how to resolve it. I see no upside to logging this to the user, but it is useful for internal users of this utility.

Reply:
I agree with that. There are other logs in this function that even emit `ERROR` level information that no URL could be generated, which I will also downgrade to debug. If they are truly errors, the function should error, but it returns None